packages/serialize/src/index.ts

```diff
@@ -107,41 +166,43 @@ if (process.env.NODE_ENV !== 'production') {
 }
 
 function handleInterpolation(
-  mergedProps /*: void | Object */,
-  registered /*: RegisteredCache | void */,
-  interpolation /*: Interpolation */
-) /*: string | number */ {
+  mergedProps: unknown | undefined,
+  registered: RegisteredCache | undefined,
+  interpolation: Interpolation | TemplateStringsArray
+): string | number {
   if (interpolation == null) {
     return ''
   }
-  if (interpolation.__emotion_styles !== undefined) {
+  const componentSelector = interpolation as ComponentSelector
+  if (componentSelector.__emotion_styles !== undefined) {
     if (
       process.env.NODE_ENV !== 'production' &&
-      interpolation.toString() === 'NO_COMPONENT_SELECTOR'
+      String(componentSelector) === 'NO_COMPONENT_SELECTOR'
     ) {
       throw new Error(
         'Component selectors can only be used in conjunction with @emotion/babel-plugin.'
       )
     }
-    return interpolation
```

Comment:
This missing return is causing the errors mentioned here: https://github.com/emotion-js/emotion/pull/2543#issuecomment-968197924

Reply:
You are right! I removed it accidentally.

It was implicitly being converted to a String, so I added an explicit cast.